drivers/storage/portworx/portworx.go

```diff
@@ -365,6 +263,118 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster)
 	setDefaultAutopilotProviders(toUpdate)
 }
 
+func (p *portworx) updateDesiredImages(toUpdate *corev1.StorageCluster, pxVersionChanged bool, release *manifest.Version) {
+	toUpdate.Status.Version = toUpdate.Spec.Version
```

Comment:
can we move this out? not something related to desired image. also resetting the desired images to empty should also be part of this function.

Reply:
I can probably rename the function, updateImageAndVersions. setting to empty is not here, because regardless what version, setting to empty should be called, hence it's outside of the function and it's always called.